### ISSUE TITLE
[Bug|Crash]: Crash when returning to main menu

### ISSUE THREAD

**Issue by SettingDust** [Yesssssman/epicfightmod#1404]
### Have you checked if a similar issue is already reported by someone else?

- [X] I checked there are no similar issues have been reported.

### Have you read the support policy?

- [X] I read it and I accept the policy.

### Are you using the latest Epic Fight and recommended Forge version?

- [X] I checked I'm using latest Epic Fight and recommended Forge version.

### Is this issue related to mod incompatibility?

- [ ] This is a mod compatibility issue and I'm aware of the problem.

### Minecraft Version

1.18.2

### What happened?

`ControllEngine` should check the player connect as well before `tick`. 
https://github.com/Yesssssman/epicfightmod/blob/1.18.2/src/main/java/yesman/epicfight/client/events/engine/ControllEngine.java#L524

In development
```
java.lang.NullPointerException: Cannot invoke "net.minecraft.client.multiplayer.ClientPacketListener.getPlayerInfo(java.util.UUID)" because the return value of "net.minecraft.client.Minecraft.getConnection()" is null
	at net.minecraft.client.player.AbstractClientPlayer.isSpectator(AbstractClientPlayer.java:52) ~[forge-1.18.2-40.2.10_mapped_official_1.18.2-recomp.jar%2380!/:?] {re:mixin,pl:runtimedistcleaner:A,re:classloading,pl:runtimedistcleaner:A}
	at yesman.epicfight.skill.BasicAttack.isExecutableState(BasicAttack.java:68) ~[epicfight-405076-4794936_mapped_official_1.18.2.jar%2388!/:18.5.14] {re:mixin,re:classloading,pl:mixin:A}
	at yesman.epicfight.skill.SkillContainer.canExecute(SkillContainer.java:241) ~[epicfight-405076-4794936_mapped_official_1.18.2.jar%2388!/:18.5.14] {re:classloading}
	at yesman.epicfight.skill.SkillContainer.sendExecuteRequest(SkillContainer.java:157) ~[epicfight-405076-4794936_mapped_official_1.18.2.jar%2388!/:18.5.14] {re:classloading}
	at yesman.epicfight.client.events.engine.ControllEngine.tick(ControllEngine.java:368) ~[epicfight-405076-4794936_mapped_official_1.18.2.jar%2388!/:18.5.14] {re:classloading,pl:runtimedistcleaner:A}
	at yesman.epicfight.client.events.engine.ControllEngine$Events.clientTickEndEvent(ControllEngine.java:492) ~[epicfight-405076-4794936_mapped_official_1.18.2.jar%2388!/:18.5.14] {re:classloading,pl:runtimedistcleaner:A}
```

In production
```
java.lang.NullPointerException: Cannot invoke "net.minecraft.client.multiplayer.ClientPacketListener.m_104949_(java.util.UUID)" because the return value of "net.minecraft.client.Minecraft.m_91403_()" is null
	at net.minecraft.client.player.AbstractClientPlayer.m_5833_(AbstractClientPlayer.java:52) ~[client-1.18.2-20220404.173914-srg.jar%23326!/:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:rei_plugin_compatibilities:B,pl:runtimedistcleaner:A}
	at yesman.epicfight.skill.BasicAttack.isExecutableState(BasicAttack.java:68) ~[EpicFight-18.5.14.jar%23142!/:18.5.14] {re:mixin,re:classloading,pl:rei_plugin_compatibilities:B,pl:mixin:A}
	at yesman.epicfight.skill.SkillContainer.canExecute(SkillContainer.java:241) ~[EpicFight-18.5.14.jar%23142!/:18.5.14] {re:classloading,pl:rei_plugin_compatibilities:B}
	at yesman.epicfight.skill.SkillContainer.sendExecuteRequest(SkillContainer.java:157) ~[EpicFight-18.5.14.jar%23142!/:18.5.14] {re:classloading,pl:rei_plugin_compatibilities:B}
	at yesman.epicfight.client.events.engine.ControllEngine.tick(ControllEngine.java:368) ~[EpicFight-18.5.14.jar%23142!/:18.5.14] {re:classloading,pl:rei_plugin_compatibilities:B,pl:runtimedistcleaner:A}
	at yesman.epicfight.client.events.engine.ControllEngine$Events.clientTickEndEvent(ControllEngine.java:492) ~[EpicFight-18.5.14.jar%23142!/:18.5.14] {re:classloading,pl:rei_plugin_compatibilities:B,pl:runtimedistcleaner:A}
```

**Comment by Yesssssman** [comment]
How can I reproduce this crash? Cause I've never seen this crash before.

**Comment by SettingDust** [comment]
> How can I reproduce this crash? Cause I've never seen this crash before.

Return to main menu when attacking.
And another point. If you think connection cloud be null when tick end. Why not start? https://github.com/Yesssssman/epicfightmod/blob/1.18.2/src/main/java/yesman/epicfight/client/events/engine/ControllEngine.java#L527

Hard to tell. Since the production is provided from the author of Dawncraft. Maybe you can get the developing pack from him and delete my fix mod.

**Comment by MetalKnight56** [comment]
Fixed in 18.5.20+, 19.5.20+, 20.6.4+
